.github/workflows/test_nightlies_on_main.yml

```diff
@@ -37,12 +35,24 @@ jobs:
         with:
           python-version: "3.12"
 
-      - name: Build package
+      - name: Setup build dependencies
+        # PIP_FLAGS: exposed via global env
+        run: |
+          source tools/github/setup_build_env.sh
+
+      - name: Build and install
+        run: |
+          pip install -vv --no-build-isolation .
+
+      - name: Setup test dependencies
+        env:
+          MINIMUM_REQUIREMENTS: 0
+          OPTIONAL_DEPS: 0
+          WITHOUT_POOCH: 0
+          # PIP_FLAGS: exposed via global env
         run: |
-          set -ex
-          source tools/github/before_install.sh
-          set -ex
-          pip install -vv --no-build-isolation .;
+          source tools/github/setup_test_env.sh
 
       - name: Run tests
-        run: source tools/github/script.sh
+        run: |
+          (cd .. && pytest --doctest-plus --showlocals --pyargs skimage)
```

Comment:
This will miss the pytest config in `pyproject.toml` but we will deal with that in a separate PR.